### DISCUSSION TITLE
Modal - Add close on route change props

### DISCUSSION THREAD

**Discussion by maclementED** [mantinedev/mantine#5851] [Feature requests]
## Problem
When using the modal manager, the modals are kept open when changing pages.

### Solution
Detect page change and close all modals. Here is the fix in Next.js.
```react
  const pathname = usePathname();

  useEffect(() => {
    closeAllModals();
  }, [pathname]);
```

I wouldn't mind implementing the feature if the feature is needed.

## Other Problem - Multiple Modals
When using the modal manager, if you open a second dialog, all states from the first dialog are cleared (probably unmounted). It's an issue when you are using modals for forms. Here is the user flow:

1. Open **Edit Entity modal**
2. Open **Edit Entity Picture modal**
3. Add a picture and close the modal
4. All info in the **Edit Entity** gets cleared.

I still managed to find workaround, but it was a pained to debuged.

**Comment by rtivital** [comment]
Routing is a part of your application, Mantine does not have access to your routing library.
